cpp/src/join/mixed_join_semi.cu

```diff
@@ -197,25 +152,33 @@ std::unique_ptr<rmm::device_uvector<size_type>> mixed_join_semi(
   auto const equality_build_conditional =
     row_comparator_conditional_build.equal_to<false>(build_nulls, compare_nulls);
   double_row_equality equality_build{equality_build_equality, equality_build_conditional};
-  make_pair_function_semi pair_func_build{};
 
-  auto iter = cudf::detail::make_counting_transform_iterator(0, pair_func_build);
+  hash_set_type set{compute_hash_table_size(build.num_rows()),
```

Comment:
```suggestion
  hash_set_type set{build.num_rows(),
```
Using `compute_hash_table_size` together with `0.5` will make the hash table with 0.25 occupancy.